libtsuba/test/rdg-part-header.cpp

```diff
@@ -0,0 +1,97 @@
+#include "RDGPartHeader.h"
+#include "katana/Result.h"
+#include "katana/URI.h"
+#include "tsuba/tsuba.h"
+
+// This test exercises the prop info lists in RDGPartHeader. The three primary
+// operations that RDGPartHeader exposes for those lists are:
+// 1. Upsert
+// 2. Remove
+// 3. Find
+//
+// This test tests all three and it also tests the PropStorageInfo state machine
+// as a side-effect of testing that the find functionality can be used to
+// transition PropStorageInfos stored within a RDGPartHeader.
+katana::Result<void>
+TestPropInfoLists(const std::string& path_to_header) {
+  katana::Uri path_to_header_uri =
+      KATANA_CHECKED(katana::Uri::Make(path_to_header));
+  tsuba::RDGPartHeader under_test =
+      KATANA_CHECKED(tsuba::RDGPartHeader::Make(path_to_header_uri));
+
+  // initial state
+  KATANA_LOG_ASSERT(under_test.node_prop_info_list().empty());
+  KATANA_LOG_ASSERT(under_test.edge_prop_info_list().size() == 1);
+  KATANA_LOG_ASSERT(under_test.find_edge_prop_info("value") != nullptr);
+  KATANA_LOG_ASSERT(under_test.find_edge_prop_info("not value") == nullptr);
+  KATANA_LOG_ASSERT(under_test.part_prop_info_list().empty());
+
+  // upserts
+  tsuba::PropStorageInfo new_edge_prop(
+      "not value", arrow::fixed_size_binary(8));
+  under_test.UpsertEdgePropStorageInfo(std::move(new_edge_prop));
+
+  tsuba::PropStorageInfo updated_edge_prop("value", arrow::large_binary());
+  under_test.UpsertEdgePropStorageInfo(std::move(updated_edge_prop));
+
+  KATANA_LOG_ASSERT(under_test.node_prop_info_list().empty());
```

Comment:
Shouldn't this be `!`?

Reply:
This is tightly coupled to rmat15 as an input. I will add some comments/asserts indicating that. But with that in mind, rmat15 has one edge property and no node properties. So this is the correct assert.

Reply:
Cool.  Comment is worthwhile.